core/dbt/cli/requires.py

```diff
@@ -277,6 +278,8 @@ def wrapper(*args, **kwargs):
                 ctx.obj["manifest"] = parse_manifest(
                     runtime_config, write_perf_info, write, ctx.obj["flags"].write_json
                 )
+            else:
+                register_adapter(runtime_config, get_mp_context())
```

Comment:
I think we need to add: 

```py
adapter = get_adapter(runtime_config)      
adapter.set_macro_context_generator(generate_runtime_macro_context)
adapter.set_macro_resolver(ctx.obj["manifest"])
```

to get the tests passing

Reply:
Thanks! Done!